routes/general.js

```diff
@@ -24,33 +25,34 @@ router.get('/register', (req, res) => {
   res.render('register');
 });
 
-router.get('/search', function (req, res) {
+router.get('/search', (req, res) => {
   res.render('search');
 });
 
-router.post('/search', (req, res) => {
+router.post('/search/:searchBy?/:searchTerm?', (req, res) => {
```

Comment:
[Route parameters](https://expressjs.com/en/guide/routing.html#route-parameters) are not what we need to use in this instance (because we're submitting a `<form>` instead of sending an AJAX request). Route parameters are only useful for capturing data that is part of the URL. For instance, The URL, `http://localhost:3001/search/title/test`, has "**title**" where the `searchBy` placeholder was, and "**test**" where the `searchTerm` placeholder was. If you tried that URL in your app then `req.params.searchBy` would get the value **title** and `req.params.searchTerm` would get the value **test**. Express does this for you.

Since we're submitting using an HTML form, the values of our form are submitted to the server within the **body** of our HTTP request (not as part of the URL). For instance, when the following form is submitted to the server, the value of the `<input>` named **title** will be sent to the server in the body of the request:

```html
<form method="post" action="/search">
  <input type="text" name="title" placeholder="search term">
  <button type="submit">Search!</button>
</form>
```

On the server, you can access that data like:

```js
router.post('/search', (req, res) => {
  let title = req.body.title;
  ...
})
```